examples/rinkeby.rs

```diff
@@ -0,0 +1,63 @@
+use ethcontract::transaction::Account;
+use ethsign::SecretKey;
+use std::env;
+use std::time::Duration;
+use web3::api::Web3;
+use web3::transports::WebSocket;
+use web3::types::H256;
+
+ethcontract::contract!("examples/truffle/build/contracts/DeployedContract.json");
+
+const RINKEBY_CHAIN_ID: u64 = 4;
+
+fn main() {
+    futures::executor::block_on(run());
+}
+
+async fn run() {
+    let account = {
+        let pk = env::var("PK").expect("PK is not set");
+        let raw_key: H256 = pk.parse().expect("invalid PK");
+        let key = SecretKey::from_raw(&raw_key[..]).expect("invalid PK");
+        Account::Offline(key, Some(RINKEBY_CHAIN_ID))
+    };
+    let infura_url = {
+        let project_id = env::var("INFURA_PROJECT_ID").expect("INFURA_PROJECT_ID is not set");
+        format!("wss://rinkeby.infura.io/ws/v3/{}", project_id)
+    };
+
+    // use a WebSocket transport to support confirmations
+    let (eloop, ws) = WebSocket::new(&infura_url).expect("transport");
+    eloop.into_remote();
```

Comment:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2FxT0xeJpnrWC4XWblEk%2Fgiphy-downsized-medium.gif&token=mind_blown" data-gitmeme-token="mind_blown"><img src="https://media2.giphy.com/media/xT0xeJpnrWC4XWblEk/giphy-downsized-medium.gif" title="Created by gitme.me with /mind_blown"/></a>

We are keeping a stupid reference to keep it alive in dex-services.